src/bscPlugin/CallExpressionInfo.ts

```diff
@@ -0,0 +1,161 @@
+import type { Expression } from '../parser/AstNode';
+import type { CallExpression, CallfuncExpression, NewExpression } from '../parser/Expression';
+import type { ClassStatement, NamespaceStatement } from '../parser/Statement';
+import { isCallExpression, isCallfuncExpression, isVariableExpression, isDottedGetExpression, isClassStatement, isNewExpression } from '../astUtils/reflection';
+import type { BrsFile } from '../files/BrsFile';
+import type { Position } from 'vscode-languageserver-protocol';
+import { util } from '../util';
+import { ParseMode } from '../parser/Parser';
+
+
+export enum CallExpressionType {
+    namespaceCall = 'namespaceCall',
+    call = 'call',
+    callFunc = 'callFunc',
+    constructorCall = 'constructorCall',
+    myClassCall = 'myClassCall',
+    otherClassCall = 'otherClassCall',
+    unknown = 'unknown'
+}
+
+//Util Class, for extracting info about an expression, which can be used in the IDE
+export class CallExpressionInfo {
```

Comment:
Everything else in the `bscPlugin` folder is event driven --- should signature help be event driven too?